src/FlatSharp.Compiler.UnityStub/NativeArray.cs

```diff
@@ -0,0 +1,46 @@
+﻿/*
+ * Copyright 2022 Unity Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Unity.Collections;
+
+public struct NativeArray<T> : IDisposable, IEnumerable<T>, IEquatable<NativeArray<T>> where T : struct
```

Comment:
Is this intended to be immutable? I was thinking it would be nice to have at least a modest implementation here so we could write some functional-ish tests.

Reply:
I can look at filling this in, although it won't be possible to match the exact implementation from Unity as it assumes it's running within a Unity Editor/Player environment.

Reply:
Not immutable, but I have added a basic implementation and a number of other tests.